howso/feature_residuals.amlg

```diff
@@ -1102,6 +1191,152 @@
 		(null)
 	)
 
+	;Helper function that takes a mapping or list that has been reduced, and expands it by readding the features from the groups that have been dropped.
+
+	;parameters:
+	; compressed_values: an assoc or list of values that need to expanded. If a list, adds back in the features from shared deviations groups that were removed
+	;  for calculations If an assoc, adds back in removed features as keys and copies the value of the primary key for each shared deviation group as its value.
+	;       e.g. {"a": 2.5, "b": 0} -> {"a": 2.5 "b":9 "c": 2.5 "d": 2.5} if features a, c, d were in a shared deviations group.
+	#!ExpandForSharedDeviations
+	(declare
+		(assoc compressed_values (null))
+		(if (size !sharedDeviationsMap)
+			(if (~ (list) compressed_values)
+				(assign (assoc
+					compressed_values (values (append compressed_values (indices !sharedDeviationsMap)) (true))
+				))
+				;if the compressed_values is not an list, it is an assoc.
+				(let
+					(assoc
+						;features that were dropped and had their values combined into their primary shared deviations feature
+						removed_features
+							(filter
+								(lambda
+									(not (contains_value
+										(indices compressed_values)
+										(current_value)
+									))
+								)
+								(indices !sharedDeviationsMap)
+							)
+					)
+					(assign (assoc
+						compressed_values
+							(append
+								compressed_values
+								;grab the values for the removed features by looking up their primary key values
+								(map
+									(lambda
+										(get compressed_values (get !sharedDeviationsMap (current_index)))
+									)
+									(zip removed_features)
+								)
+							)
+					))
+				)
+			)
+		)
+		compressed_values
+	)
+
+	;Helper function to prep shared feature deviations, by taking in a list of features and their list of residuals,
+	; combining all of the residuals under the primary feature key of the group of the shared deviations, and removing the features
+	; from the features list whose residuals have been combined into another features.
+	;parameters:
+	; features_list: A features list
+	; feature_residuals_lists: A features residuals whose lists are in the order of the features_list
+	#!PrepSharedDeviations
+	(declare
+		(assoc
+			features_list (assoc)
```

Comment:
rename this to `features` since that's what is passed in anyway, and is already a list of features, features_list sounds like it's a list of lists of features.